### ISSUE TITLE
Excel WEB Addin disappears (crashed) in <Excel on the web> if javascript code recreates table in parallel session 

### ISSUE THREAD

**Issue by ASamSam** [OfficeDev/office-js#1803]
Having two parallel sessions in a single file on SharePoint and Addin activated in both:
in the first session, the addin recreates (deletes and then adds) the worksheet, then creates a table there and fills it with data. at the 'add rows' step the addin in second session disappears without any errors reported in the console.

## Expected Behavior
Addin in second session remains stable

## Current Behavior
With javascript console, breakpoints and some 'await context.sync();' we found, that recreating worksheet, adding table and defining table headers passes normally, but after 'table.rows.add(0 , data['data']);' the addin in second session disappears.

## Steps to Reproduce, or Live Example
Reproducing is simple: 
open same doc on two pages, 
then activate script lab in both (no matter if script itself is executed or not). 
Then click 'Run' at the executed one. 
* Link to live example: [https://gist.github.com/ASamSam/ad3c7717988628d6a361018205cef0ea](url)

after clicking the button in executed script - script lab in second page will disappear. If several addons (our own in my eample) activated - all will disappear

## Context
we are developing Excel addin able to be used by several people for the single file.
seems, at the moment file data is loaded by addin - all parallel sessions will be killed

## Your Environment
Excel on the Web, parallel sessions (no matter - in same browser or on different PC),
file shared with SharePoint 

## Useful logs
there are no any error log in javascript console at the crash moment

link on stackoverflow:
[https://stackoverflow.com/questions/66891572/excel-web-addin-disappears-crashed-if-javascript-code-in-parallel-session-recr?noredirect=1#comment118661572_66891572](https://stackoverflow.com/questions/66891572/excel-web-addin-disappears-crashed-if-javascript-code-in-parallel-session-recr?noredirect=1#comment118661572_66891572)

**Comment by jipinshi** [comment]
Thanks reporting the issue, we are investigating

**Comment by gingerjia** [comment]
@ASamSam, I'm working on the investigation and not sure if it's kind of by design. 
Since I tried the same steps but co-authoring  between Excel online and Excel Win32, I found after running script to add table and row at Excel online, at Win32 it will show up a yellow bar to reminder you to apply a new change. If you click yes to accept the new change, then there will be a refresh and taskpane closed at Win32 also. I will go on looking into the issue to see if anyway can improve.

And to unblock your side first, can you call range.values to replace table.rows.add API in your Add-in? It has the same effect to add new table rows and won't hit taskpane close issue. Here is sample gist for your reference: https://gist.github.com/gingerjia/99a30ccad5402e50bc8f977e5b286016

Any other comments or questions are welcomed! Thanks!

**Comment by ASamSam** [comment]
thanks for the proposal, going to apply it.

I have a couple of cases when addin stops working in excel online, but can`t figure out how to prepare an example or get some logs:
1. i have opened excel document and activated addin. After some time of inactivity (half of an hour? hour?) addin remains shown, but when I do click on it - it instantly disappears.
2.  i have opened excel document and activated addin - put PC to sleep mode (for a night). after activating and enabling the browser page - addin disappears in a second without any actions from me.

What are possible reasons for that? javascript common runtime is used.
how I can make the addin more stable? i would like to have it working all the time the document is active
clearly, questions are little bit unrelated to the initial one, yet also refers to unexpected addin stops. and not precise enough to report as individual ones

**Comment by ASamSam** [comment]
@gingerjia , I did apply the proposed workaround and found another issue, even worse I would say: #1813

**Comment by gingerjia** [comment]
"  i have opened excel document and activated addin. After some time of inactivity (half of an hour? hour?) addin remains shown, but when I do click on it - it instantly disappears.
    i have opened excel document and activated addin - put PC to sleep mode (for a night). after activating and enabling the browser page - addin disappears in a second without any actions from me. "

@ASamSam, for your above questions, it's by design as if no connectivity after 5 min~  and no-interactivity after 10 hours, then will trigger Excel silent reload, which will close the taskpane. If it's a very strong requirement from your side, feel free to label it as  prod requirement and our PM will prioritize the work! Thanks!

For your original issue, maybe you can share more about your detail scenario to us if possible then to see if any available workaround to unblock your work?

**Comment by ASamSam** [comment]
@gingerjia , thanks a lot for the description. I have a feeling that I didn't see it in the specification. Ok, so it is not a result of a crash.
we are trying to use Excel as an interface allowing several users to change some data altogether.
data is loaded from our database through addin and stored back at some moment.
so, the 'loading' moment when the document is totally recreated (pages are removed and then added back, then tables filled) is more or less rare. I`ll see if I can refresh data without recreating pages

**Comment by gingerjia** [comment]
@ASamSam please see the workaround provided at https://github.com/OfficeDev/office-js/issues/1813. 
If this can unblock your work now, that would be great. Thanks very much for reporting so great questions!
